frontend/src/components/test-history.js

```diff
@@ -33,334 +31,317 @@ import {
-    comparisonResults: PropTypes.array
-  };
+const WEEKS = {
+  '1 Week': 0.25,
+  '2 Weeks': 0.5,
+  '1 Month': 1.0,
+  '2 Months': 2.0,
+  '3 Months': 3.0,
+  '5 Months': 5.0
+};
 
-  constructor (props) {
-    super(props);
-    this.state = {
-      columns: [{title: 'Result', cellFormatters: [expandable]}, 'Source', 'Exception Name', 'Duration', 'Start Time'],
-      rows: [getSpinnerRow(5)],
-      results: [],
-      cursor: null,
-      pageSize: 10,
-      page: 1,
-      totalItems: 0,
-      totalPages: 0,
-      isEmpty: false,
-      isError: false,
-      isFieldOpen: false,
-      isOperationOpen: false,
-      isDropdownOpen: false,
-      onlyFailures: false,
-      historySummary: null,
-      dropdownSelection: '1 Week',
-      lastPassedDate: 'n/a',
-      filters: Object.assign({
-        'result': {op: 'in', val: 'passed;skipped;failed;error;xpassed;xfailed'},
-        'test_id': {op: 'eq', val: props.testResult.test_id},
-        'component': {op: 'eq', val: props.testResult.component},
-        // default to filter only from 1 weeks ago to the most test's start_time.
-        'start_time': {op: 'gt', val: new Date(new Date(props.testResult.start_time).getTime() - (0.25 * 30 * 86400 * 1000)).toISOString()}
-      }, props.filters),
-      comparisonResults: this.props.comparisonResults
-    };
-    // filter on env by default
-    if (props.testResult.env) {
-      this.state.filters['env'] = {op: 'eq', val: props.testResult.env};
-    }
-    this.refreshResults = this.refreshResults.bind(this);
-    this.onCollapse = this.onCollapse.bind(this);
-  }
+const RESULT_STATES = {
+  'passed': 'passes',
+  'failed': 'failures',
+  'error': 'errors',
+  'skipped': 'skips',
+  'xfailed': 'xfailures',
+  'xpassed': 'xpasses'
+};
 
-  refreshResults = () => {
-    this.getHistorySummary();
-  };
+const TestHistoryTable = (props) => {
+  const {
+    comparisonResults,
+    filters,
+    testResult
+  } = props;
 
-  onCollapse (event, rowIndex, isOpen) {
-    const { rows } = this.state;
+  const [rows, setRows] = useState([getSpinnerRow(5)]);
+  const [pageSize, setPageSize] = useState(10);
+  const [page, setPage] = useState(1);
+  const [totalItems, setTotalItems] = useState(0);
+  const [isEmpty, setIsEmpty] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [onlyFailures, setOnlyFailures] = useState(false);
+  const [historySummary, setHistorySummary] = useState();
+  const [dropdownSelection, setDropdownSelection] = useState('1 Week');
+  const [filtersState, setFiltersState] = useState({});
 
-    // lazy-load the result view so we don't have to make a bunch of artifact requests
-    if (isOpen) {
-      rows[rowIndex + 1].cells = [{
-        title: <ResultView hideTestHistory={true} hideSummary={false} hideTestObject={false} testResult={rows[rowIndex].result}/>
-      }];
+  useEffect(() => {
+    let env_filter = {};
```

Comment:
Nitpick: Could be `const` since we are not re-assigning object but changing it's props